app/presenters/dashboard_presenter.rb

```diff
@@ -32,6 +32,10 @@ def campaigns
     current_user.campaigns
   end
 
+  def ungreeted_students(course)
+    course.students.ungreeted.length
```

Comment:
Even with the `includes(:students)` moved to the right place above, this still triggering a query for each course, because `ungreeted` is an ActiveRecord scope.

For the case where there are no students, doing it as an array operation makes it 2x faster and removes the N+!, but I haven't tested what it's like in more like production conditions where most courses have many students. `course.students.reject(&:greeted).count`

But I think finding a way to do the calculation of the count in that main query (or a separate query for the counts per course, all in one) will be the ideal way to do it, if we can figure out how.